### DISCUSSION TITLE
[Proposal] Skip Once for a block of code, or function.

### DISCUSSION THREAD

**Discussion by Nathaniel25** [dotnet/csharplang#6223] [Language Ideas]
A Skip Once for a block of code.

Instead of having an If statement catch whether a variable hasn't been assigned yet, you skip it just once.
In my unity project I am creating a mesh and in my context, I would like to clear it's data before I create it, even though of course it hasn't been created.

They are two different functions and are within a loop, I would like to avoid If statements as much as possible, so I thought it would be quite cool to have a thing that would skip a block of code. Once its called again it would run, and possibly having a reset and variable number of times it skips before it runs. 

I figured this would save on some computation, and have many use cases.

**Comment by erik-kallen** [comment]
Perhaps you can achieve what you want with a static constructor?

**Comment by HaloFour** [comment]
The only way the compiler could achieve such a thing would be to emit the equivalent of `if` statements within your loop, so there'd be nothing gained there.  Given it would require some kind of syntax to denote the "skipped" block I doubt there'd be any verbosity savings either.

Why not put this logic at the end of the loop rather than at the beginning?  That way you don't have to be concerned with attempting to skip it, it would always happen before the next iteration.

**Reply by TonyValenti** [reply]
Can you post the code you are hoping to have a better solution for?
